source/codegen/templates/service_helpers.mako

```diff
@@ -286,7 +304,10 @@ ${initialize_standard_input_param(function_name, parameter)}\
       if (${size} > 0) {
           ${parameter_name}.resize(${size}-1);
       }
-%     elif underlying_param_type in ['ViAddr', 'ViInt32', 'ViUInt32', 'ViUInt16']:
+%     elif service_helpers.is_output_array_that_needs_coercion(parameter):
+      std::vector<${underlying_param_type}> ${parameter_name}(${size});
+## uInt32 requires cast because of int vs long in that typedef vs uint32_t
+%     elif underlying_param_type in ['ViAddr', 'ViInt32', 'ViUInt32', 'ViUInt16', 'uInt32']:
```

Comment:
Ugh. I think you should leave this one the way you have it. I think we'd want to address this case in a generalized conversion implementation, but I don't think it makes sense to try to come up with a one-off for it.